File: calculator/application/controller/discount.go
package controller

type Discount struct {
	Percentage   float32
	ValueInCents int
}

func NewProductDiscount(percentage float32, value_in_cents int) *Discount {
	return &Discount{
		Percentage:   percentage,
		ValueInCents: value_in_cents,
	}
}

File: product-list/src/Api/Domain/Entity/Discount.php
<?php


namespace Api\Domain\Entity;


class Discount implements \JsonSerializable
{
    private float $percentage;

    private int $valueInCents;

    /**
     * Discount constructor.
     * @param float $percentage
     * @param int $valueInCents
     */
    public function __construct(float $percentage, int $valueInCents)
    {
        $this->percentage = $percentage;
        $this->valueInCents = $valueInCents;
    }

    public function jsonSerialize():array
    {
        return [
            'percentage' => round($this->percentage, 2),
            'value_in_cents' => $this->valueInCents
        ];
    }
}